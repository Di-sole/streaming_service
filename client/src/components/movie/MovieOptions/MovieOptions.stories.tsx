import { Meta, StoryObj } from '@storybook/react';
import MovieOptions from './MovieOptions';
import './MovieOptions.module.scss';

import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport';

const meta: Meta<typeof MovieOptions> = {
    title: 'Information/Options',
    component: MovieOptions,
    parameters: {
        viewport: {
            viewports: INITIAL_VIEWPORTS,
            default: 'desktop',
        },
    },
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19' }}>
                <Story />
            </div>
        ),
    ],
};

export default meta;

type Story = StoryObj<typeof MovieOptions>;

export const FullScreen: Story = {};

export const MobileScreen: Story = {
    parameters: {
        viewport: {
            defaultViewport: 'iphonex',
        },
    },
};
