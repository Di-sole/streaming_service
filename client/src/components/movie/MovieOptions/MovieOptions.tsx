import styles from './MovieOptions.module.scss';
import TextBadge from '../../UI/badges/TextBadge/TextBadge';
import { useTranslation } from 'next-i18next';

const MovieOptions = () => {
    const { t } = useTranslation('movie');
    return (
        <div className={styles.container} data-testid={'options'}>
            <div className={styles.container__block}>
                <div className={styles.container__title}>{t('languages')}</div>
                <div className={styles.container__value}>{t('langVariant')}</div>
            </div>

            <div className={styles.container__block}>
                <div className={styles.container__title}>{t('subtitles')}</div>
                <div className={styles.container__value}>{t('subVariant')}</div>
            </div>

            <div className={styles.container__block}>
                <div className={[styles.container__title, styles.container__title_full].join(' ')}>
                    {t('quality')}
                    <span className={styles.container__quality}> {t('qualityDisclaimer')}</span>
                </div>

                <div className={[styles.container__title, styles.container__title_mobile].join(' ')}>
                    {t('qualityAdaptive')}
                </div>

                <div className={styles.container__badgesContainer}>
                    {['FullHD', 'HD', '1080', '720'].map((display) => (
                        <div key={display} className={styles.container__badge}>
                            <TextBadge text={display} />
                        </div>
                    ))}
                </div>
            </div>
        </div>
    );
};

export default MovieOptions;
