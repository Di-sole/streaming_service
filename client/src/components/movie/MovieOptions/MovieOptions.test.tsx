import { render, screen } from '@testing-library/react';
import MovieOptions from './MovieOptions';

describe('MOVIEOPTIONS SNAPSHOTS TESTS', () => {
    test('MovieOptions should not be changed', () => {
        render(<MovieOptions />);

        const div = screen.getByTestId('options');
        expect(div).toMatchSnapshot();
    });
});
