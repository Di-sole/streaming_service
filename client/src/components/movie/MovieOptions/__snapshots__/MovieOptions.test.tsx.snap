// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOVIEOPTIONS SNAPSHOTS TESTS MovieOptions should not be changed 1`] = `
<div
  class="container"
  data-testid="options"
>
  <div
    class="container__block"
  >
    <div
      class="container__title"
    >
      languages
    </div>
    <div
      class="container__value"
    >
      langVariant
    </div>
  </div>
  <div
    class="container__block"
  >
    <div
      class="container__title"
    >
      subtitles
    </div>
    <div
      class="container__value"
    >
      subVariant
    </div>
  </div>
  <div
    class="container__block"
  >
    <div
      class="container__title container__title_full"
    >
      quality
      <span
        class="container__quality"
      >
         
        qualityDisclaimer
      </span>
    </div>
    <div
      class="container__title container__title_mobile"
    >
      qualityAdaptive
    </div>
    <div
      class="container__badgesContainer"
    >
      <div
        class="container__badge"
      >
        <div
          class="badge"
          data-testid="textBadge"
        >
          FullHD
        </div>
      </div>
      <div
        class="container__badge"
      >
        <div
          class="badge"
          data-testid="textBadge"
        >
          HD
        </div>
      </div>
      <div
        class="container__badge"
      >
        <div
          class="badge"
          data-testid="textBadge"
        >
          1080
        </div>
      </div>
      <div
        class="container__badge"
      >
        <div
          class="badge"
          data-testid="textBadge"
        >
          720
        </div>
      </div>
    </div>
  </div>
</div>
`;
